### ISSUE TITLE
[Vitess11] [VReplication] VReplication doesn't copy timestamp column with default current_timestamp

### ISSUE THREAD

**Issue by tokikanno** [vitessio/vitess#8759]
#### Overview of the Issue
VReplication doesn't copy timestamp column with default current_timestamp

#### Reproduction Steps

* Create a table which has a timestamp column with DEFAULT CURRENT_TIMESTAMP setting

```sql
create table toki_test3 (
	id varchar(3) NOT NULL,
	created TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
	PRIMARY KEY(`id`)
);
```

* Insert several rows into it

```sql
insert into src.toki_test3 values ('AAA', NULL);
insert into src.toki_test3 values ('BBB', NULL);
```

```
select * from src.toki_test3;
+-----+---------------------+
| id  | created             |
+-----+---------------------+
| AAA | 2021-09-02 11:41:53 |
| BBB | 2021-09-02 11:41:54 |
+-----+---------------------+
```

* Start MoveTables workflow on created table

```sh
vtctlclient MoveTables -source src -tables toki_test3 Create dest.test
```

* Verify copied table

```
select * from dest.toki_test3;
+-----+---------------------+
| id  | created             |
+-----+---------------------+
| AAA | 2021-09-02 11:45:13 |
| BBB | 2021-09-02 11:45:13 |
+-----+---------------------+
```

The `created` value is not the same as origin table, seems VReplication didn't copy the values of `created` column from origin table.

#### Binary version

Vitess V11 release

#### Related slack threads

- https://vitess.slack.com/archives/CMKTCUYNQ/p1630500506059800
- https://vitess.slack.com/archives/CMKTCUYNQ/p1630554910072800

#### Possible reason

This issue may caused by PR #8129 

The copied table schema has an extra info `DEFAULT_GENERATED` (the origin table didn’t)

```
desc toki_test3;
+---------+------------+------+-----+-------------------+-------------------+
| Field   | Type       | Null | Key | Default           | Extra             |
+---------+------------+------+-----+-------------------+-------------------+
| id      | varchar(3) | NO   | PRI | NULL              |                   |
| created | timestamp  | NO   |     | CURRENT_TIMESTAMP | DEFAULT_GENERATED |
+---------+------------+------+-----+-------------------+-------------------+
```

Via these codes (merged from #8129 )
https://github.com/vitessio/vitess/blob/6b52e982ccdf1c20b1b11d84693e5d57da6b46b6/go/vt/vttablet/tabletmanager/vreplication/vreplicator.go#L296
https://github.com/vitessio/vitess/blob/332223cc2b1e917284ccdc2be29d47f55a747f49/go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go#L340

The `created` timestamp column seems will be ignored during replication because Vitess think it is a generated column

**Comment by tokikanno** [comment]
From the master server binlog, the insert value of `created` column has been recorded correctly

```
BINLOG '
AUgwYRNGPDQBOwAAAEbWbAcAANUXAAAAAAEABnBpbmtvaQAKdG9raV90ZXN0MwACDxEDAwAAAMPv
loU=
AUgwYR5GPDQBLAAAAHLWbAcAANUXAAAAAAEAAgAC//wDQUFBYTBIAT7Dumc=
'/*!*/;
### INSERT INTO `pinkoi`.`toki_test3`
### SET
###   @1='AAA'
###   @2=1630554113
```

**Comment by tokikanno** [comment]
Also, if change the create table sql into 

```sql
create table toki_test3 (
	id varchar(3) NOT NULL,
	created TIMESTAMP NOT NULL DEFAULT 0,
	PRIMARY KEY(`id`)
);
```

Table will be replicated by Vitess correctly

**Comment by tokikanno** [comment]
Maybe I found the reason

The copied table schema has an extra info DEFAULT_GENERATED (the origin table didn’t)

```
desc toki_test3;
+---------+------------+------+-----+-------------------+-------------------+
| Field   | Type       | Null | Key | Default           | Extra             |
+---------+------------+------+-----+-------------------+-------------------+
| id      | varchar(3) | NO   | PRI | NULL              |                   |
| created | timestamp  | NO   |     | CURRENT_TIMESTAMP | DEFAULT_GENERATED |
+---------+------------+------+-----+-------------------+-------------------+
```

Via these codes (merged from #8129 )
https://github.com/vitessio/vitess/blob/6b52e982ccdf1c20b1b11d84693e5d57da6b46b6/go/vt/vttablet/tabletmanager/vreplication/vreplicator.go#L296
https://github.com/vitessio/vitess/blob/332223cc2b1e917284ccdc2be29d47f55a747f49/go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go#L340

The created column seems will be ignored during replication because Vitess think it is a generated column

**Comment by mattlord** [comment]
This was resolved via https://github.com/vitessio/vitess/pull/8763 and https://github.com/vitessio/vitess/pull/8796. 

Thank you so much for the great issue and PR, @tokikanno !
